Update to Serana Dialogue Addon

ddbb07 opened the pull request:
- Added Patch Messages

ddbb07 committed:
Update to Serana Dialogue Addon

- Added Patch Messages

sibir-ine commented:
Some of the patches are purely optional so it might be best to not include messages for them. I added a few a while back that probably also count for this, so, if @MacSplody thinks it's appropriate, do you mind removing them as well & only keeping the ones that are necessary? I'm thinking patches like the ones for the armor, but some others might fall under this umbrella too. Also, could you remove the alsoUseX message while your at it? I originally added it, but I don't really think the author's recommendation warrants the message anymore. No problem if you don't want to, I can in another PR.

MacSplody commented:
> Some of the patches are purely optional so it might be best to not include messages for them. I added a few a while back that probably also count for this, so, if @MacSplody thinks it's appropriate, do you mind removing them as well & only keeping the ones that are necessary? I'm thinking patches like the ones for the armor, but some others might fall under this umbrella too.

I'm leaning more towards leaving optional patches out, but there's no real harm in adding them, so I have no objections.
Up to you @ddbb07 on whether you agree with @sibir-ine's point or not.

MacSplody commented on masterlist.yaml:
```diff
@@ -9363,12 +9363,30 @@ plugins:
+      - <<: *patchProvided
+        subs: [ 'Serana Dialogue Edit' ]
+        condition: 'active("Serana Dialogue Edit.esp") and not active("SDA_SDE Patch.esp")'
+      - <<: *patchProvided
+        subs: [ 'Wintersun - Faiths of Skyrim' ]
+        condition: 'active("Wintersun - Faiths of Skyrim.esp") and not active("Complete Alchemy & Cooking Overhaul.esp") and(not active("SDA Wintersun Patch.esp") or not checksum("SDA Wintersun Patch.esp", A0EB4F89))'
+      - <<: *patchProvided
+        subs: [ 'Wintersun and CACO' ]
+        condition: 'active("Wintersun - Faiths of Skyrim.esp") and active("Complete Alchemy & Cooking Overhaul.esp") and(not active("SDA Wintersun Patch.esp") or not checksum("SDA Wintersun Patch.esp", 2F889D8A))'
```
My suggestion from Discord.

```suggestion
      - <<: *patchProvided
        subs: [ 'Wintersun - Faiths of Skyrim' ]
        condition: 'active("Wintersun - Faiths of Skyrim.esp") and not active("Complete Alchemy & Cooking Overhaul.esp") and not active("SDA Wintersun Patch.esp")'
      - <<: *patchProvided
        subs: [ 'Wintersun and CACO' ]
        condition: 'active("Wintersun - Faiths of Skyrim.esp") and active("Complete Alchemy & Cooking Overhaul.esp") and not active("SDA Wintersun Patch.esp")'
```

With a separate message for the wrong patch being used.

```yaml
  - name: 'SDA Wintersun Patch.esp'
    msg:
      - <<: *useVersion
        subs: [ 'Complete Alchemy & Cooking Overhaul' ]
        condition: 'active("Complete Alchemy & Cooking Overhaul.esp") and checksum("SDA Wintersun Patch.esp", A0EB4F89)'
```
If you prefer your way that's fine, I just think it's a bit clearer this way.

ddbb07 replied:
Im sorry for not fixing this earlier i have just not had alot of time, i'll mark it as a draft until im done

MacSplody replied:
> Im sorry for not fixing this earlier i have just not had alot of time, i'll mark it as a draft until im done

There's generally no rush when submitting changes, so don't worry about it.

ddbb07 committed:
Removed AlsoUse Message

ddbb07 committed:
Update

- Removed checksum checks for patch messages.
- Added useVersion to Wintersun Patch

ddbb07 committed:
Update

- Changed optional patches to alsoUseX

ddbb07 commented:
I changed optional patches from patchProvided to alsoUseX, if you don't like the change just leave a comment.

sibir-ine commented on masterlist.yaml:
```diff
@@ -9361,19 +9361,40 @@ plugins:
+  - name: 'SDA Wintersun Patch.esp'
+    msg:
+      - <<: *useVersion
+        subs: [ 'Wintersun and CACAO Patch' ]
```
This fits the message a bit better:
```suggestion
        subs: [ 'Complete Alchemy & Cooking Overhaul' ]
```
Either that or:
```suggestion
        subs: [ 'CACO' ]
```
Whichever you prefer.

sibir-ine reviewed (changes requested):
Other than my suggestion, my only concern is that if someone has several of these mods, it may feel a bit cluttered to have multiple messages for optional patches they can't dismiss without installing them. However, this may be addressed in a future version of LOOT as described in loot/loot#1083, so this is fine for now.

ddbb07 committed:
Update masterlist.yaml

Co-authored-by: Sibir <69141501+sibir-ine@users.noreply.github.com>

sibir-ine commented:
Thanks!
